makePileup / PileupUtil helper

ryan-williams opened the pull request:
reduce test boilerplate around creating pileups

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/hammerlab/guacamole/533)
<!-- Reviewable:end -->

timodonnell commented:
This overall seems more complicated to me than the code it replaces, especially the stuff involving `implicit`

ryan-williams commented:
A bunch of tests instantiate `Pileup`s to do various things. They are not intending to test anything related to `Pileup` construction; in particular, they always want to grab the `ContigSequence` for the `ContigName` that the `Pileup` is being created on. Moreover, they have an ambient `ReferenceBroadcast` laying around from which to get said `ContigSequence`.

Instrumenting test `Suite`s to reduce mindless boilerplate at the sub-test-case level (some tests make multiple pileups) reduces complexity overall; two possible ways of doing so are:

1. declare an abstract `def reference: ReferenceBroadcast` in `pileup.Util` that will be used in `Util.makePileup`, and implemented in each `Suite`.
2. have `makePileup` take the `ReferenceBroadcast` it needs `implicit`ly, and require `Suite`s to declare their ambient `ReferenceBroadcast`s to be `implicit`.

The former has two main drawbacks relative to the latter:
- it's not possible for a `Suite` to use one `ReferenceBroadcast` in one place and another elsewhere.
- `Suite`s must name their `ReferenceBroadcast`s the same thing (`reference` in this example).

The `implicit` route fixes both of these problems, and is equally as statically checked, which is to say basically completely: missing or extraneous `implicit ReferenceBroadcast`s will be compile-time errors.

So I disagree that this is more complex; it puts test-Pileup-creation in one place that is then reused many dozens of times, and a quick skim of the diff shows a lot of red: redundant code that has been eliminated. 

Lots of our tests are hard to follow for being walls of text due to poor decomposition of boilerplate setup code like this.

timodonnell commented on src/test/scala/org/hammerlab/guacamole/commands/SomaticStandardCallerSuite.scala:
```diff
@@ -238,7 +242,7 @@ class SomaticStandardCallerSuite
       )
 
-    val tumorPileup = Pileup(tumorReads, "chr1", 2, simpleReference.getContig("chr1"))
+    val tumorPileup = makePileup(tumorReads)
```
from looking at this it's not obvious to me what position the pileup will be at. Would be good to include those arguments in these places `"chr1"` and `2`

ryan-williams replied:
sure, agreed this is a bridge too far, removing those default args

ryan-williams commented:
OK, I reworked things to make `reference` part of the `Util` mix-in. I also moved two other `TestUtil` methods involved in creating pileups into `pileup.Util`.
